src/uwtools/drivers/driver.py

```diff
@@ -3,29 +3,41 @@
 """
 
 from abc import ABC, abstractmethod
+from collections.abc import Mapping
 from typing import Optional
 
-from uwtools import config_validator
 from uwtools.logger import Logger
+from uwtools.scheduler import BatchScript
+
+from uwtools import config_validator
 
 
 class Driver(ABC):
     """
     An abstract class representing drivers for various NWP tools.
     """
 
-    def __init__(self, config_file: str, log: Optional[Logger] = None):
+    def __init__(
+        self,
+        config_file: str,
+        log: Optional[Logger] = None,
+        dry_run: bool = False,
+        outfile: Optional[str] = None,
+    ):
         """
         Initialize the driver.
         """
+
         self.log = log if log is not None else Logger()
+        self._dry_run = dry_run
         self._config_file = config_file
+        self._outfile = outfile
         self._validate()
 
     # Public methods
 
     @abstractmethod
-    def batch_script(self) -> None:
+    def batch_script(self, job_resources: Mapping) -> BatchScript:
```

Comment:
The `job_resources` argument is called, if I'm not mistaken, `platform_resources` at the call site. Any reason for two names for (I think) the same concept?